src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -103,6 +119,16 @@ public static void RegisterInstance(VisualStudioInstance instance)
                 throw new ArgumentNullException(nameof(instance));
             }
 
+            if (instance.DiscoveryType == DiscoveryType.DotNetSdk)
+            {
+                // Since there can be multiple DotNet SDKs on a machine
+                // the MSBuild environment variable are likely not set.
+                // To work around this problem we configure the necessary
+                // variables for a DotNet SDK deployment of MSBuild.
+                // Workaround for https://github.com/dotnet/docfx/issues/1752
```

Comment:
This is not really working around a bug. Certainly not a docfx one.

I think it would be simpler to explain that normally the dotnet cli sets up these environment variables with msbuild is invoked via `dotnet`, but we are using msbuild dlls directly and therefore need to mimic that.